### PR TITLE
Heap

### PR DESCRIPTION
**Pull request by xiaqian1999**

### CODE REVIEW

File: xiaqianZhang/assignments/heaps/lc295/lc295.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
+# Problem Statement #
+
+# Design a class to calculate the median of a number stream. The class should have the following two methods:
+
+#   1. insertNum(int num): stores the number in the class
+#   2. findMedian(): returns the median of all numbers inserted in the class
+
+# If the count of numbers inserted in the class is even, the median will be the average of the middle two numbers.
+
```
Comment (knrodriguez) [inline xiaqianZhang/assignments/heaps/lc295/lc295.py:9]: Missing approach and edge cases.

File: xiaqianZhang/assignments/heaps/lc295/lc295.py [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    if self.heapList is []:
+      return None
+
+    #start of by getting the median index
+    med = self.currentSize // 2
+    med1 = self.currentSize // 2 - 1
+    if self.currentSize % 2 == 0:
+      return (self.heapList[med] + self.heapList[med1]) / 2
+    else:
+      return self.heapList[med]
```
Comment (knrodriguez) [inline xiaqianZhang/assignments/heaps/lc295/lc295.py:39]: Are you sure this solution works as intended?

File: xiaqianZhang/assignments/heaps/lc295/lc295.py [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+  medianOfAStream.insert_num(1)
+  # [1, 3]
+  print("The median is: " + str(medianOfAStream.find_median()))
+  # 2
+  medianOfAStream.insert_num(5)
+  print("The median is: " + str(medianOfAStream.find_median()))
+  medianOfAStream.insert_num(4)
+  # index 4 -1 // 2, so the parent is 5, swap becomes [1, 3, 4, 5]
+  print("The median is: " + str(medianOfAStream.find_median()))
+
```
Comment (knrodriguez) [inline xiaqianZhang/assignments/heaps/lc295/lc295.py:84]: Missing custom test cases.

File: xiaqianZhang/assignments/twopointers/lc1/lc1.py [hunk 4]
```diff
@@ -3,6 +3,44 @@
...
+edge case: if the array is empty, return null
+
+initliaze two variables, two pointer that one starts from index 0, another starts at the end of the array
+  while the left pointer is smaller than the right pointer:
+    initliaze a currentSum that is the sum of the start and end
+    check if the currentSum is:
+      equal as the target: return the index of start and end
+      smaller than the target: move the start index rightwards because the currentSum is too small
+      less than 0: move the end index leftwards since the currentSum is too large
+'''
```
Comment (knrodriguez) [inline xiaqianZhang/assignments/twopointers/lc1/lc1.py:19]: Clear and concise approach. Missing edge cases.

**Review by knrodriguez — APPROVED** [review]

**Comment by knrodriguez** [comment]
Great job revisiting previous patterns and re-acclimating yourself with the material.
